pbkdf2-check.c

```diff
@@ -108,15 +108,35 @@ int pbkdf2_check(char *password, char *hash)
 		return (FALSE);
 	}
 
+#ifdef RAW_SALT
+	char *rawSalt;
+
+	if ((rawSalt = malloc(strlen(salt) + 1)) == NULL) {
+		fprintf(stderr, "Out of memory\n");
+		return FALSE;
+	}
+
+	saltlen = base64_decode(salt, rawSalt);
+	if (saltlen < 1) {
+		return (FALSE);
+	}
+
+	free(salt);
+	salt = strdup(rawSalt);
```

Comment:
if `rawSalt` contains binary data, the `strdup()` might fail (i.e. stop copying at a binary 0). Can that happen?